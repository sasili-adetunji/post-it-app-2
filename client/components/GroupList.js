import React from 'react';
import { CardHeader, CardTitle, Card } from 'material-ui/Card';
import Toggle from 'material-ui/Toggle';
import { List } from 'material-ui/List';
import RaisedButton from 'material-ui/RaisedButton';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import PostItActions from '../actions/PostItActions';
import AddMember from './AddMember';
import CreateGroup from './CreateGroup';
import MessageBox from './MessageBox';
import Group from './Group';
import PostItStore from '../stores/PostItStore';


/**
 * Grouplist component with createGroup,MessageBox and AddMember component
 *
 * @class GroupList
 * @extends {React.Component}
 */
class GroupList extends React.Component {

  /**
   * Creates an instance of GroupList.
   * @param {any} props
   * @memberof GroupList
   */
  constructor(props) {
    super(props);
    this.state = {
      toggledAdd: false,
      showAdd: false,
      showCreate: false,
      showMessage: false,
      toggledCreate: false,
      toggledMessage: false
    };
    this.handleToggleAdd = this.handleToggleAdd.bind(this);
    this.handleToggleCreate = this.handleToggleCreate.bind(this);
  }

  /**
   * controls the toggle of addMember components
   *
   * @memberof GroupList
   */
  handleToggleAdd() {
    this.setState({
      toggledAdd: !this.state.toggledAdd
    });
  }

  /**
   * controls the toggle for creatMember components
   *
   * @memberof GroupList
   */
  handleToggleCreate() {
    this.setState({
      toggledCreate: !this.state.toggledCreate
    });
  }

  /**
   * The markup for the groupList components
   *
   * @returns {void}
   * @memberof GroupList
   */
  render() {
    const groupNodes = this.props.groups.map((group, i) => {
      return (
        <Group group={group} key={i} />
      );
    });
    return (
      <div className="bottomMargin">
        <MuiThemeProvider>
          <Card>
            <List >
              <CardTitle title="My Groups" />
              {groupNodes}
            </List>
           <Toggle
            name="createGroup"
            label="Create Group"
            defaultToggled={this.state.toggledCreate}
            onToggle={this.handleToggleCreate}
          />
          <Toggle
            name="addMember"
            label="Add Member"
            defaultToggled={this.state.toggledAdd}
            onToggle={this.handleToggleAdd}
          />
          {this.state.toggledAdd ? this.state.showAdd = true :
          this.state.showAdd = false}
          {this.state.showAdd ? <AddMember /> : ''}

          {this.state.toggledCreate ? this.state.showCreate = true :
          this.state.showCreate = false}
          {this.state.showCreate ? <CreateGroup /> : ''}
          </Card>
        </MuiThemeProvider>
        <div>
          
          
        </div>
      </div>
    );
  }
}


export default GroupList;
