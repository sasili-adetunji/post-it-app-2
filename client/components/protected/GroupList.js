import React from 'react';
import CreateGroup from './CreateGroup';
import PostItStore from '../../stores/PostItStore';
import Group from './Group';
import API from '../../Api';


/**
 * cretes grouplist components
 *
 * @class GroupList
 * @extends {React.Component}
 */
class GroupList extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      groups: PostItStore.getGroupsUser(),
    };
  }

  onChange() {
    this.setState({
      groups: PostItStore.getGroupsUser(),
    });
  }

  componentWillReceiveProps() {
    API.getUserGroups();
  }

  render() {
    let header = null;
    if (this.props.grou.length < 1) {
      header = (<h4> No Group yet </h4>);
    } else {
      header = (<h4 className="card-header"> My groups </h4>);
    }
    const groupNodes = this.props.grou.map((group, i) => {
      return (
        <Group group={group} key={i} />
      );
    });
    return (
      <div>
        <div className="headerlist"> {header} </div>
        <div className="groupList">
        {groupNodes}
        </div>
      </div>
    );
  }
}
export default GroupList;
