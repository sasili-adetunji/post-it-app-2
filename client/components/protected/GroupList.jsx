import React from 'react';
import PostItStore from '../../stores/PostItStore';
import Group from './Group';

/**
 * @description Displays a list of groups a user belongs to
 * 
 * @function GroupList
 * 
 * @returns {JSX} list of groups a user belongs
 */

const GroupList = () => {
    let header = null;
    if (PostItStore.getGroupsUser().length < 1) {
      header = (<h4> No Group yet </h4>);
    } else {
      header = (<h4 className="card-header"> My groups </h4>);
    }
    const groupNodes = PostItStore.getGroupsUser().map((group, i) => {
      return (
        <Group group={group} key={i} />
      );
    });
    return (
      <div>
        <div className="headerlist"> {header} </div>
        <div className="groupList">
          {groupNodes}
        </div>
      </div>
    );
}
export default GroupList;
