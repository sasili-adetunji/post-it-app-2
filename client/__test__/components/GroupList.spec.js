import React from 'react';
import { shallow, render } from 'enzyme';
import GroupList from '../../components/protected/GroupList.jsx';

describe('GroupList components', () => {
  const component = shallow(<GroupList />);

  it('should match snapshot test', () => {
    expect(component).toMatchSnapshot();
  });
  it('should render', () => {
    expect(component).toBeDefined();
  });
  it('Should contain three div', () => {
    expect(component.find('div').length).toEqual(3);
  });
  it('should recieve props', () => {
    expect(Object.keys(component.props()).length).toBeGreaterThan(0);
  });
});
