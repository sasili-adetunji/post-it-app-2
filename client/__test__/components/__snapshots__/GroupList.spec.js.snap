// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroupList components should match snapshot test 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <div
      className="headerlist"
    >
       
      <h4>
         No Group yet 
      </h4>
       
    </div>
    <div
      className="groupList"
    />
  </div>,
  "nodes": Array [
    <div>
      <div
        className="headerlist"
      >
         
        <h4>
           No Group yet 
        </h4>
         
      </div>
      <div
        className="groupList"
      />
    </div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_context": Object {},
    "_element": <GroupList />,
    "_instance": GroupList {
      "context": Object {},
      "onChange": [Function],
      "props": Object {},
      "refs": Object {},
      "state": Object {
        "groups": Array [],
        "loggedInUser": Array [],
        "message": Array [],
        "readUsers": Array [],
        "selectedGroup": Array [],
        "showAddUser": false,
        "showCreateGroup": false,
        "user": Array [],
        "users": Array [],
      },
      "updater": Updater {
        "_renderer": [Circular],
      },
    },
    "_newState": null,
    "_rendered": <div>
      <div
        className="headerlist"
      >
         
        <h4>
           No Group yet 
        </h4>
         
      </div>
      <div
        className="groupList"
      />
    </div>,
    "_rendering": false,
    "_updater": Updater {
      "_renderer": [Circular],
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <GroupList />,
}
`;
